### ISSUE TITLE
Status of the report header is not updated after the expense gets paid

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#52596]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.62-4
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:**
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @dubielzyk-expensify 
**Slack conversation** (hyperlinked to channel name): [Expensify-Expense](https://expensify.slack.com/archives/C06ML6X0W9L/p1731558967756909)

## Action Performed:
1. Submit a report in a workspace
2. As approver approve and pay the expense
3. Verify the report header "next steps" as a submitter
## Expected Result:
Should get updated
## Actual Result:
Not updated but inside the expense it gets updated
## Workaround:
unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details>

https://github.com/user-attachments/assets/d09c35f7-fdf3-4a9f-af6e-1b3944ba1466

![CleanShot 2024-11-14 at 10 23 59@2x](https://github.com/user-attachments/assets/b2096e5a-5ddc-4516-856f-b8a1f6de8e97)

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by Shahidullah-Muffakir** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
**MoneyReportHeader** is not updating the the iou message
### What is the root cause of that problem?
we are using '-1' as if reportID is not there, it is causing the issue.
### What changes do you think we should make in order to solve the problem?
in **MoneyReportHeader** replace "-1", with ""

**Comment by Shahidullah-Muffakir** [comment]
When the approver pay the expense using `PayMoneyRequest` endpoint, in the frontend ,in the approver side the `reportNextStep_reportID` gets updated. 
 in the employee side Pusher update the report_reprotID and its message prop also gets updated, that is why we see the updated message **paid $250 elsewhere** inside the expense
 but the reportNextStep_reportID, does not gets updated by Pusher which has the message we are using in `MoneyReportHeader` 
https://github.com/Expensify/App/blob/a6f1348d08a441f4f9b48a8ef05e5e95c96b53d7/src/components/MoneyReportHeader.tsx#L69

when user reopen this page, the OpenReport endpoint is triggered, the BE send us the updated value for reportNextStep_reportID, then the header gets updated and showing **"No further action required!".**

so to fix this issue, when `PayMoneyRequest` end point is called, BE should also send updates for `reportNextStep_reportID` key to the frontend.

**Comment by JmillsExpensify** [comment]
@mountiny thoughts on that last comment?

**Comment by mountiny** [comment]
Yep we can do that, we do something like that in other commands.

**Comment by JmillsExpensify** [comment]
Waiting on internal volunteers

**Comment by JmillsExpensify** [comment]
Still waiting on an internal volunteer. I'm going to make a hot pick for migrate.

**Comment by JmillsExpensify** [comment]
Still waiting on internal volunteer

**Comment by JmillsExpensify** [comment]
Still waiting on internal volunteer for hot pick.

**Comment by maddylewis** [comment]
@JmillsExpensify - when you have a sec, could you write the responses to these prompts? and then ill add this to the `HOT PICKS` section of the Retain update today 👍 

https://stackoverflowteams.com/c/expensify/questions/20462

**Comment by JmillsExpensify** [comment]
@trjExpensify Yes, I agree with that. I'm actually not super sure where we're at with the report titles in auth migration, which could theoretically affect this right?

**Comment by neil-marcellini** [comment]
@trjExpensify would you please create a separate issue for the report title changing and make sure we have agreement from the team on that UX change? You're welcome to assign me.

**Comment by trjExpensify** [comment]
>I'm actually not super sure where we're at with the report titles in auth migration

@thienlnam has a [draft PR](https://github.com/Expensify/Auth/pull/13522) that he's going to pick up again now after the holidays. 

>which could theoretically affect this right?

Hm, not sure really. Moving them to auth would benefit using the "proper" default report title set on the workspace, not the one with the reportID. I don't quite see why that necessitates preventing updating the title from what we have to something different upon being marked as paid though. Perhaps @thienlnam you can give us an opinion on that? 

>@trjExpensify would you please create a separate issue for the report title changing and make sure we have agreement from the team on that UX change? You're welcome to assign me.

Sure, will do. I'll wait for Jack to confirm on the above re moving report title generation to auth first though.

**Comment by thienlnam** [comment]
Agreed with Tom, I don't think this is related to the report formulas in auth unless the report title specifically has the report state in the formula

**Comment by trjExpensify** [comment]
Kewl, issue here: https://github.com/Expensify/App/issues/54814

**Comment by JmillsExpensify** [comment]
@neil-marcellini with the Web PR on production, can we close this issue? Or is another PR coming?

**Comment by JmillsExpensify** [comment]
Bump on that last question when you return from OOO @neil-marcellini

**Comment by neil-marcellini** [comment]
Yeah it should be all done
